src/tracing/mod.rs

```diff
@@ -57,21 +58,18 @@ impl<P: EthereumProvider + Send + Sync + Clone> Tracer<P> {
 
                 let parity_builder = inspector.into_parity_builder();
 
-                // Extract the base fee from the transaction based on the transaction type
-                let base_fee = match tx.transaction_type.and_then(|typ| typ.try_into().ok()) {
-                    Some(TxType::Legacy) | Some(TxType::Eip2930) => tx.gas_price,
-                    Some(TxType::Eip1559) => tx
-                        .max_fee_per_gas
-                        .map(|fee| fee.saturating_sub(tx.max_priority_fee_per_gas.unwrap_or_default())),
-                    _ => return Err(EthereumDataFormatError::TransactionConversionError.into()),
-                };
-
                 let transaction_info = TransactionInfo {
                     hash: Some(tx.hash),
                     index: tx.transaction_index,
                     block_hash: tx.block_hash,
                     block_number: tx.block_number,
-                    base_fee,
+                    base_fee: Some(
+                        env.env
+                            .block
+                            .basefee
+                            .try_into()
+                            .map_err(|err: FromUintError<u128>| TransactionError::Tracing(err.into()))?,
```

Comment:
can you extract this into `let basefee =...`? would be more readable

Reply:
Done